core/src/main/java/io/questdb/griffin/FunctionFactory.java

```diff
@@ -76,6 +76,10 @@ default boolean isBoolean() {
         return false;
     }
 
+    default boolean isRuntimeConstant() {
```

Comment:
The same method exists on function and these 2 seems to be not aligned. Not everywhere where you set Factory to be runtime constant the resulting function is runtime constant. I'd suggest using another name, say isRuntimeConstantFactory and / or align the functions with factories to avoid future confusion.

Some functions can still be constants even though produced by non-const factories like sum of const is a const.

Reply:
this is not a constant:

```select sum(1) from tab```

Reply:
I meant

```
select dateadd('d', -1, now()), count(*) from tab
```

Reply:
All functions produced by a FunctionFactory that is runtime constant should either be constant or runtime constant. However, the opposite is not true, in the the case of dateadd the factory is not runtime constant because the whether the function is constant or not will depend on the arguments.